File: src/03-21-2022b/Solution.java
package mar21b;

public class Solution {

  private static String convertZigZag(String s, int numRows) {
    String zigzag = "";
    for (int i = 0; i < numRows; i ++) {
      int incr = 0, cond = 0;
      while (cond - i < s.length()) {
        if (i != 0 && i != numRows - 1 && cond - i > 0) {
          zigzag = zigzag + s.charAt(cond - i);
        }
        if (cond + i < s.length()) {
          zigzag = zigzag + s.charAt(cond + i);
        }
        cond = (2 * (++incr)) * (numRows - 1);
      }
    }
    return zigzag;
  }

  public static void main(String[] args) {
    String s = "PAYPALISHIRING";
    int numRows = 3;

    System.out.println("Zig Zag conversion of " + s + " in " + numRows + " rows is: " + convertZigZag(s, numRows));
  }
}

File: src/03-21-2022b/solution.js
const getZigZag = (line, numRows) => {
  let zigZag = '';
  for (let i = 0; i < numRows; i++) {
    let incr = 0;
    let comp = (2 * incr) * (numRows - 1);
    while (comp < line.length) {
      if (i != 0 && i != numRows - 1 && comp - i > 0) {
        zigZag += line[comp - i];
      }
      if (comp + i < line.length) {
        zigZag += line[comp + i];
      }
      comp = (2 * ++incr) * (numRows - 1);
    }
  }
  return zigZag;
};

line = 'PAYPALISHIRING';
numRows = 4;
console.log(`Conversion of ${line} to zigzag in ${numRows} rows is: ${getZigZag(line, numRows)}`);